devops/docker/CIDockerfile

```diff
@@ -7,8 +7,9 @@ ENV DOCKER_SHA256_x86_64 340e0b5a009ba70e1b644136b94d13824db0aeb52e09071410f35a9
 
 RUN apt-get update && \
     apt-get install -y flake8 make virtualenv ccontrol libpython2.7-dev \
-            libffi-dev libssl-dev libyaml-dev python-pip curl git &&\
-    apt-get clean
+            libffi-dev libssl-dev libyaml-dev python-pip curl git npm &&\
+    apt-get clean && \
+    npm --global install html_codesniffer
```

Comment:
@saffronsnail, we discussed this internally last week and reached the consensus that we are not comfortable with an `npm install --global` invocation.  Instead, please run `npm install` locally, commit the resulting `package.json` and `package-lock.json`, and then here do:

```suggestion
    npm ci
```

npm history suggests that even this measure isn't foolproof (e.g., npm/cli#2701, npm/cli#4460), but it's our bare minimum in general and should be adequate in the development and CI environments.

(We may choose to do some follow-up work to minimize the number and size of the toolchains we install in these containers, but that's out of the scope of this pull request.)